core/storage/story/gae_models.py

```diff
@@ -77,22 +77,21 @@ def _trusted_commit(
                     cmd: str. Unique command.
                 and then additional arguments for that command.
         """
-        super(StoryModel, self)._trusted_commit(
-            committer_id, commit_type, commit_message, commit_cmds)
-
         committer_user_settings_model = (
             user_models.UserSettingsModel.get_by_id(committer_id))
         committer_username = (
             committer_user_settings_model.username
             if committer_user_settings_model else '')
 
         story_commit_log_entry = StoryCommitLogEntryModel.create(
-            self.id, self.version, committer_id, committer_username,
+            self.id, self.version + 1, committer_id, committer_username,
             commit_type, commit_message, commit_cmds,
             feconf.ACTIVITY_STATUS_PUBLIC, False
         )
         story_commit_log_entry.story_id = self.id
         story_commit_log_entry.put_async()
+        super(StoryModel, self)._trusted_commit(
```

Comment:
Can you explain why this was moved here? Just curious, and wondering whether we need to do it elsewhere too.

Reply:
(answer for this and the comment below) I didn't test with the others, but when I manually tested with stories, the CommitLogEntry model that was created didn't increment version (didn't create an additional entry each time I did an edit) although it was incremented in _trusted_commit() in base_model. So, I thought it might be because of the put_async() that caused some kind of race condition, as entries in multiples of 2 were created (every other edit had a corresponding CommitLogEntry). So, I put the _trusted_commit() line after the CommitLogEntry line and explicitly incremented the version when creating the CommitLogEntry model without changing the variable itself (as this increment was done in _trusted_commit() ) and with this config, it started recording correctly.

Reply:
Thanks for the explanation. That's not enough, though, I'm afraid -- we need to understand exactly (from an analytical point of view) why the change works and why the previous version of the code didn't work, and ensure that all models conform to a structure/pattern that works. Otherwise the current "recording correctly" might be a fluke. Could you investigate a bit more until you're absolutely sure about what the problem is?

Also, let me know if you need help, I'm happy to talk about this.